### PR TITLE
better error on union of `StrawberryList`

### LINKED ISSUE #2591: Provide better error for Union of scalars.
This keeps coming from questions on discord
```py
union: Union[List[int], List[str]]
```
gives something like
```console
  File "/root/backend/env/lib/python3.10/site-packages/strawberry/schema/name_converter.py", line 99, in from_union
    assert hasattr(type_, "_type_definition")
AssertionError

The above exception was the direct cause of the following exception:
  File "/root/backend/env/lib/python3.10/site-packages/graphql/type/definition.py", line 811, in fields
    raise cls(f"{self.name} fields cannot be resolved. {error}") from error
TypeError: Query fields cannot be resolved
```

### PR DESCRIPTION
**Pull request by nrbnlulu**
fixes #2591

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release covers an edge case where
```python
some_field: "Union[list[str], SomeType]]"
```
would not give a nice error.
fixes [#2591](https://github.com/strawberry-graphql/strawberry/issues/2591)


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to ניר for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: tests/schema/test_union.py [hunk 1]
```diff
@@ -662,3 +662,30 @@ class Query:
...
+
+
+@pytest.mark.parametrize(
+    "annotation", ("int", "str", "float", "list[str]", "List[str]")
+)
+def test_raises_on_union_of_scalars(annotation: str):
+    # using forward refs because they deduced later
+    # catching all edge cases.
+    if annotation == "list[str]" and sys.version_info < (3, 9, 0):
+        return
```
Comment (bellini666) [inline tests/schema/test_union.py:674]: I think you can do `pytest.skip()` in here

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Release type: patch
+
+This release covers an edge case where
+```python
+some_field: "Union[list[str], SomeType]]"
+```
+would not give a nice error.
```
Comment (bellini666) [inline RELEASE.md:7]:
```suggestion
This release covers an edge case where the following would not give a nice error.
```python
some_field: "Union[list[str], SomeType]]"
```

```

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+fixes [#2591](https://github.com/strawberry-graphql/strawberry/issues/2591)
```
Comment (bellini666) [inline RELEASE.md:8]:
```suggestion
Fixes [#2591](https://github.com/strawberry-graphql/strawberry/issues/2591)
```

**Review by bellini666 — CHANGES_REQUESTED** [review]

File: tests/schema/test_union.py (hunk 1 above)
Comment (nrbnlulu) [reply tests/schema/test_union.py:674]: hmm that is just for one parameterization though... You know how to do that?

File: tests/schema/test_union.py [hunk 4]
```diff
@@ -662,3 +662,29 @@ class Query:
...
+
+
+@pytest.mark.parametrize(
+    "annotation", ("int", "str", "float", "list[str]", "List[str]")
+)
+def test_raises_on_union_of_scalars(annotation: str, monkeypatch):
+    # using forward refs because they deduced later
+    # catching all edge cases.
+    if annotation == "list[str]" and sys.version_info < (3, 9, 0):
+        pytest.skip("list has not __class_getitem__ prior to py39")
```
Comment (skilkis) [inline tests/schema/test_union.py:674]: This can be placed in the parametrize on line 668 using `pytest.param("list[str", marks=pytest.mark.skipif(...))` 😄

**Review by patrick91 — APPROVED** [review]
